tests/runtime/service_generation/test_primitives.py

```diff
@@ -38,9 +37,41 @@ def test_to_primitive_signature_union_dm():
     assert to_primitive_signature(
         signature={"name": Union[SampleInputType, str]},
         primitive_data_model_types=["sample_lib.data_model.SampleInputType"],
+    ) == {"name": Union[SampleInputType, str]}
+
+
+def test_to_primitive_signature_unsupported_type_in_union():
+    assert to_primitive_signature(
+        signature={"name": Union[SampleInputType, str]},
+        primitive_data_model_types=[],
+    ) == {"name": str}
+
+
+def test_to_primitive_signature_no_dm_primitives_in_union():
+    assert to_primitive_signature(
+        signature={"name": Union[SampleInputType, SampleOutputType, str]},
+        primitive_data_model_types=[],
+    ) == {"name": str}
+
+
+def test_to_primitive_signature_multiple_primitives_in_union():
+    """We have the first arg as a supported DM arg, and the last as
+    a supported primitive arg. We return the first supported DM arg"""
+    assert to_primitive_signature(
+        signature={"name": Union[SampleInputType, SampleOutputType, str]},
+        primitive_data_model_types=["sample_lib.data_model.SampleInputType"],
     ) == {"name": SampleInputType}
 
 
+def test_to_primitive_signature_multiple_no_dm_primitives_in_union():
+    """We have 1 dm arg that's not supported, and 2 primitive args. We
+    return the first primitive arg"""
+    assert to_primitive_signature(
+        signature={"name": Union[SampleInputType, str, int]},
+        primitive_data_model_types=[],
+    ) == {"name": str}
```

Comment:
🤔 These last two test cases are kinda making me question this whole design with "primitiveness" of data models.

Given the move to `@task`s to specify the minimum set of required args, I bet this whole "is a DM a primitive of the domain" idea goes away entirely, as we'll probably flip the logic of
- `include an api if all inputs are domain primitive` to 
- `include an api if it contains the required parameters for this task`

And I also bet it's more important to consider type unions with non-protobuffable things, like 
```
def run(foo: MyCoolDataModel | pandas.DataFrame | spark.DataFrame)
```

Given that, I might lean more towards the behavior of `handle_primitives_in_union` to be a simple removal of any types in the union that cannot be serialized to protobuf. 
Thoughts?

Reply:
This might be directly in opposition to @gabe-l-hart's thoughts on `Union[supported_type, unsupported_type]`, though I think I'm redefining `unsupported_type` simply as anything that is not serializable